motan-core/src/main/java/com/weibo/api/motan/serialize/KryoSerialization.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright 2009-2016 Weibo, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.weibo.api.motan.serialize;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.weibo.api.motan.codec.Serialization;
+import com.weibo.api.motan.core.extension.SpiMeta;
+
+import java.io.IOException;
+
+/**
+ * kryo 序列化
+ * 只支持java, 性能比hessian2更好
+ *
+ * @author liaojia1
+ */
+@SpiMeta(name = "kryo")
+public class KryoSerialization implements Serialization {
+
+    private final Kryo kryo;
+
+    public KryoSerialization() {
+        this.kryo = new Kryo();
+        kryo.setReferences(false);
+    }
+
+    @Override
+    public byte[] serialize(Object data) throws IOException {
+        Output output = new Output(16 * 1024, Integer.MAX_VALUE);
+        kryo.writeObject(output, data);
+        return output.toBytes();
```

Comment:
空实现指的是？我看代码里应该有flush操作https://github.com/EsotericSoftware/kryo/blob/master/src/com/esotericsoftware/kryo/io/Output.java#L195

```
/** Writes the buffered bytes to the underlying OutputStream, if any. */
	public void flush () throws KryoException {
		if (outputStream == null) return;
		try {
			outputStream.write(buffer, 0, position);
			outputStream.flush();
		} catch (IOException ex) {
			throw new KryoException(ex);
		}
		total += position;
		position = 0;
	}
```

文档里也注明了需要做flush：https://github.com/EsotericSoftware/kryo#io

```
Because Output buffers when writing to an OutputStream, be sure to call flush() or close() after writing is complete so the buffered bytes are written to the underlying stream.
```